excelerator/imports/convert.js

```diff
@@ -312,140 +324,31 @@ function addToCache(dataCache, toUri, row, i, jobData, valFunc) {
-                    matchObj.fromArea = b.objectAreaUnique.value;
-                if (b.subjectAreaUnique)
-                    matchObj.area = b.subjectAreaUnique.value;
-            } else {
-                if (b.subjectAreaUnique)
-                    matchObj.fromArea = b.subjectAreaUnique.value;
-                if (b.objectAreaUnique)
-                    matchObj.area = b.objectAreaUnique.value;
-            }
-            return matchObj;
-        })
-        return matches;
-    } catch (e) {
-        console.log(e)
-        throw e;
-    }
-}
-
-function getOverlapStatements(fromUri, isReverse, predUri, linksetUri) {
-    var filterExp = `${isReverse ? '?obj' : '?sub'} = <${fromUri}>`
-    var query = `PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
-PREFIX geo: <http://www.opengis.net/ont/geosparql#>
-PREFIX dct: <http://purl.org/dc/terms/>
-prefix dbp: <http://dbpedia.org/property/>
-PREFIX nv: <http://qudt.org/schema/qudt#numericValue>
-PREFIX qu: <http://qudt.org/schema/qudt#unit>
-PREFIX m2: <http://qudt.org/schema/qudt#SquareMeter>
-PREFIX c: <http://www.opengis.net/ont/geosparql#sfContains>
-PREFIX f: <http://www.opengis.net/ont/geosparql#Feature>
-PREFIX l: <${linksetUri}>
-
-SELECT ?sub ?subArea ?subUnit ?obj ?objArea ?objUnit ?intersectArea ?intersectUnit
-WHERE {
-    ?s dct:isPartOf l: ;
-        rdf:subject ?sub;
-        rdf:predicate <${predUri}> ;
-        rdf:object ?obj .
-    ?sub dbp:area [ nv: ?subArea ;
-            qu: ?subUnit ] .
-    ?obj dbp:area [ nv: ?objArea ;
-            qu: ?objUnit ] .    
-    ?intersect a f: ;
-        dbp:area [ nv: ?intersectArea ;
-            qu: ?intersectUnit ] .     
-    ?overlap1 dct:isPartOf l: ;
-        rdf:subject ?sub ;
-        rdf:predicate c: ;
-            rdf:object ?intersect .    
-    ?overlap2 dct:isPartOf l: ;
-        rdf:subject ?obj  ;
-        rdf:predicate c: ;
-            rdf:object ?intersect .
-    FILTER(${filterExp})
-}`;
-
-    try {
-        var result = getQueryResults(query);
+        });
+        console.log(result);
         var json = JSON.parse(result.content);
-        var bindings = json.results.bindings;
-        var matches = bindings.map(b => ({
-            from: {
-                uri: isReverse ? b.obj.value : b.sub.value,
-                unit: isReverse ? b.objUnit.value : b.subUnit.value,
-                area: isReverse ? b.objArea.value : b.subArea.value,
-            },
-            to: {
-                uri: isReverse ? b.sub.value : b.obj.value,
-                unit: isReverse ? b.subUnit.value : b.objUnit.value,
-                area: isReverse ? b.subArea.value : b.objArea.value,
-            },
-            overlap: {
-                unit: b.intersectUnit.value,
-                area: b.intersectArea.value,
-            },
+        var objectArea = +json.meta.featureArea;
+        var outputObjects = json.overlaps;
+
+        var matches = outputObjects.map(outObj => ({
```

Comment:
I'm not sure but if statements like fromArea: +objectArea are doing accumulations then I don't think they are necessary because unique targets should only occur once in the output of the API and accumulations should happen internally.

Reply:
`+json.meta.featureArea` simply converts the value in the JSON from a string to a number. I asked for the JSON to come back as a number, not string, but that wasn't simple at the time. Hopefully in the future we can drop the `+` but it will work with numbers or strings.

Reply:
Thanks I did not know that.